File: src/app/models/Ingredient.ts

export interface Ingredient {
  id: string
  name: string
  ingredientType: string
  img: string
  itemGroups: string[]
  price?: string
}

export type Ingredients = Ingredient[]

export interface IngredientType {
  id: string
  name: string
  price: string
  selectType: 'single' | 'multiple'
  img: string
}

export type IngredientTypes = IngredientType[]

File: WebAPI/Models/Ingredient.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WebAPI.Data
{
  public class Ingredient
  {
    [Required]
    [Key]
    public string id { get; set; }
    [Required]
    public string name { get; set; }
    [Required]
    public List<string> itemGroups { get; set; }
    [Required]
    public string ingredientType { get; set; }
    public string img { get; set; }
  }
}
